website/static/js/share/utils.js

```diff
@@ -87,6 +134,52 @@ var maybeQuashEvent = function(event) {
     }
 };
 
+var updateFilter = function(vm, filter, required){
+    if (required && vm.requiredFilters.indexOf(filter) === -1){
+        vm.requiredFilters.push(filter);
+    } else if (vm.optionalFilters.indexOf(filter) === -1 && !required){
+        vm.optionalFilters.push(filter);
+    }
+    search(vm);
+};
+
+var removeFilter = function(vm, filter){
+    var reqIndex = vm.requiredFilters.indexOf(filter);
+    var optIndex = vm.optionalFilters.indexOf(filter);
+    if (reqIndex > -1) {
+        vm.requiredFilters.splice(reqIndex, 1);
+    }
+    if (optIndex > -1) {
+        vm.optionalFilters.splice(optIndex, 1);
+    }
+    search(vm);
+};
+
+var arrayEqual = function(a, b) {
+    return $(a).not(b).length === 0 && $(b).not(a).length === 0
+};
+
+var loadStats = function(vm){
+    vm.statsLoaded(false);
+
+    m.request({
```

Comment:
Functions that make requests should return the promise, which may be useful for the caller.